db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -268,10 +268,10 @@ Status DBImpl::FlushMemTableToOutputFile(
           error_handler_.SetBGError(io_s,
                                     BackgroundErrorReason::kManifestWriteNoWAL);
         }
-      } else if (total_log_size_ > 0 || !log_io_s.ok()) {
-        error_handler_.SetBGError(io_s, BackgroundErrorReason::kFlush);
       } else {
-        // If the WAL is empty, we use different error reason
+        // If WAL sync is successful (either WAL size is 0 or there is no IO
+        // error), all the other SST file write errors will be set as
+        // kFlushNoWAL.
```

Comment:
The ```kFlushNoWAL``` and ```kManifestWriteNoWAL``` have become a little bit of a misnomer, since there may be a non-empty WAL. It can be addressed later I think, but something to keep in mind.

Reply:
Yeah, I have created a task to refactor it with reason+context, then we can clearly separate them with reason==flush, context==SST file or Manifest file. Will work on it later.